models/src/node.rs

```diff
@@ -0,0 +1,213 @@
+    pub node_uid: String,
+}
+
+fn node_resource_name(node_selector: &BottlerocketNodeSelector) -> String {
+    format!("brn-{}", node_selector.node_name)
+}
+
+#[async_trait]
+/// A trait providing an interface to interact with BottlerocketNode objects. This is provided as a trait
+/// in order to allow mocks to be used for testing purposes.
+pub trait BottlerocketNodeClient: Clone + Sync + Send + Sized {
+    /// Create a BottlerocketNode object for the specified node.
+    async fn create_node(
+        &self,
+        selector: &BottlerocketNodeSelector,
+    ) -> Result<BottlerocketNode, BottlerocketNodeError>;
+    async fn update_node_status(
+        &self,
+        selector: &BottlerocketNodeSelector,
+        status: &BottlerocketNodeStatus,
+    ) -> Result<(), BottlerocketNodeError>;
+    async fn update_node_spec(
+        &self,
+        selector: &BottlerocketNodeSelector,
+        spec: &BottlerocketNodeSpec,
+    ) -> Result<(), BottlerocketNodeError>;
+}
+
+#[derive(Clone)]
+/// Concrete implementation of the `BottlerocketNodeClient` trait. This implementation will almost
+/// certainly be used in any case that isn't a unit test.
+pub struct K8SBottlerocketNodeClient {
+    k8s_client: kube::client::Client,
+}
+
+impl K8SBottlerocketNodeClient {
+    pub fn new(k8s_client: kube::client::Client) -> Self {
+        K8SBottlerocketNodeClient { k8s_client }
+    }
+}
+
+#[async_trait]
+impl BottlerocketNodeClient for K8SBottlerocketNodeClient {
+    async fn create_node(
+        &self,
+        selector: &BottlerocketNodeSelector,
+    ) -> Result<BottlerocketNode, BottlerocketNodeError> {
+        let br_node = BottlerocketNode {
+            metadata: ObjectMeta {
+                name: Some(node_resource_name(&selector)),
+                owner_references: Some(vec![OwnerReference {
+                    api_version: "v1".to_string(),
+                    kind: "BottlerocketNode".to_string(),
+                    name: selector.node_name.clone(),
+                    uid: selector.node_uid.clone(),
+                    ..Default::default()
+                }]),
+                ..Default::default()
+            },
+            spec: BottlerocketNodeSpec::default(),
+            ..Default::default()
+        };
+
+        Api::namespaced(self.k8s_client.clone(), constants::NAMESPACE)
+            .create(&PostParams::default(), &br_node)
+            .await
+            .map_err(|err| Box::new(err) as Box<dyn std::error::Error>)
+            .context(CreateBottlerocketNode {
+                selector: selector.clone(),
+            })?;
+
+        Ok(br_node)
+    }
+
+    async fn update_node_status(
+        &self,
+        selector: &BottlerocketNodeSelector,
+        status: &BottlerocketNodeStatus,
+    ) -> Result<(), BottlerocketNodeError> {
+        let br_node_patch = json!({
+            "apiVersion": constants::API_VERSION,
+            "kind": "BottlerocketNode",
+            "status": status
+        });
```

Comment:
Will this be at any point its own serializable struct? Probably with `serde_json`?